cli/src/pcluster/schemas/imagebuilder_schema.py

```diff
@@ -147,6 +148,14 @@ def no_coexist_role_policies(self, data, **kwargs):
         if self.fields_coexist(data, ["instance_role", "additional_iam_policies"], **kwargs):
             raise ValidationError("InstanceRole and AdditionalIamPolicies can not be configured together.")
 
+    @validates_schema
+    def no_coexist_instance_profile(self, data, **kwargs):
+        """Validate that instance_profile and additional_security_groups or instance_role do not co-exist."""
+        if self.fields_coexist(data, ["instance_profile", "additional_iam_policies"], **kwargs):
+            raise ValidationError("InstanceProfile and AdditionalIamPolicies can not be configured together.")
+        if self.fields_coexist(data, ["instance_profile", "instance_role"], **kwargs):
```

Comment:
shall we also check the co-exist of  "instance_role" and "additional_iam_policies"? I think we can use fields_coexist function including all the fields:
```
if self.fields_coexist(data, ["instance_profile", "instance_role", ""additional_iam_policies"], **kwargs):
```

Reply:
Cool! Changed to include all fields